pkg/build/registry/buildconfig/strategy.go

```diff
@@ -40,9 +49,20 @@ func (strategy) AllowUnconditionalUpdate() bool {
 }
 
 // PrepareForCreate clears fields that are not allowed to be set by end users on creation.
-func (strategy) PrepareForCreate(ctx apirequest.Context, obj runtime.Object) {
+func (s strategy) PrepareForCreate(ctx apirequest.Context, obj runtime.Object) {
 	bc := obj.(*api.BuildConfig)
 	dropUnknownTriggers(bc)
+
+	// legacy buildconfig api does not apply default pruning values, to maintain
+	// backwards compatibility.
+	if !s.legacy {
```

Comment:
you mean two separate impls? (with one presumably embedding the other and then overriding this specific logic?  I can..